### ISSUE TITLE
Enhancement: PSR-2 style named migrations.

### ISSUE THREAD

**Issue by davidjeddy** [yiisoft/yii2#11096]
**Problem:**
 + Migration classes are named using a different convention than other classes in the framework

**Example:**

    yii migrate/mark 150101_185401                      # using timestamp to specify the migration
    yii migrate/mark "2015-01-01 18:54:01"              # using a string that can be parsed by strtotime()
    yii migrate/mark m150101_185401_create_news_table   # using full name
    yii migrate/mark 1392853618                         # using UNIX timestamp

Suggestion is to allow PSR2 style names:

    yii migration/mark m150101185401CreateNewsTable   # PSR2 compliant file and class name

**Impact:**
 + Linters will not fail when encountering migration classes
 + Global consistency in class naming convention

**Comment by cesarnicola** [comment]
I think I can do this change, PSR2 should be the new default naming format, right?

**Comment by samdark** [comment]
Yes.

**Comment by davidjeddy** [comment]
Thank you both very much!

**Comment by cesarnicola** [comment]
Will use "M" instead of "m" as the prefix to follow the upper-camel convention.

Also, should the create command raise a warning/error if the name does not met the convention?

Thanks!

**Comment by Faryshta** [comment]
I still think its better  to have the migration names to be something like `console\migrations\CreatePost` and have a dependency property to know which migrations should be executed first.

```php
namespace common\migrations;

class CreatePost extends Migration
{
    public function up()
    {
    }

    public function down()
    {
    }

    $depends = [
        'common\\migrations\\Init';
        'common\\migrations\\CreatePostCategory';
    ];
}
```

**Comment by samdark** [comment]
@Faryshta this way last migration out of 100 will have 99 dependencies. That's no good.

**Comment by samdark** [comment]
@cesarnicola yes, a warning would be good. Also don't forget namespaces.

**Comment by cesarnicola** [comment]
Thanks, working on it.

**Comment by Faryshta** [comment]
> this way last migration out of 100 will have 99 dependencies.

Nested migrations like in a fixture or an asset. If migration Z depends on Y and X, then it will check the dependencies of Y and X.

The last migration might depend on a 100 migrations but the code will be much simplier.

**Comment by klimov-paul** [comment]
Solution proposed at #12511

**Comment by klimov-paul** [comment]
Resolved by commit 8aa0e85
